src/main/scala/mesosphere/marathon/core/health/impl/AppHealthCheckActor.scala

```diff
@@ -0,0 +1,172 @@
+package mesosphere.marathon
+package core.health.impl
+
+import akka.actor.{ Actor, Props }
+import akka.event.EventStream
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.core.event.InstanceHealthChanged
+import mesosphere.marathon.core.health.impl.AppHealthCheckActor._
+import mesosphere.marathon.core.health.{ Health, HealthCheck }
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.state.{ PathId, Timestamp }
+
+import scala.collection.generic.Subtractable
+import scala.collection.mutable
+
+object AppHealthCheckActor {
+  case class ApplicationKey(appId: PathId, version: Timestamp)
+  case class InstanceKey(applicationKey: ApplicationKey, instanceId: Instance.Id)
+
+  def props(eventBus: EventStream): Props = Props(new AppHealthCheckActor(eventBus))
+
+  case class AddHealthCheck(appKey: ApplicationKey, healthCheck: HealthCheck)
+  case class RemoveHealthCheck(appKey: ApplicationKey, healthCheck: HealthCheck)
+
+  case class PurgeHealthCheckStatuses(hc: Seq[(InstanceKey, HealthCheck)])
+  case class HealthCheckStatusChanged(
+      appKey: ApplicationKey,
+      healthCheck: HealthCheck, health: Health)
+
+  trait InstanceHealthChangedNotifier {
+    def notify(applicationKey: ApplicationKey, instanceId: Instance.Id,
+      healthiness: Option[Boolean]): Unit
+  }
+
+  class AppHealthCheckProxy extends StrictLogging {
```

Comment:
Can we document this?:

- Used internally by …
- NOT thread safe

Reply:
done